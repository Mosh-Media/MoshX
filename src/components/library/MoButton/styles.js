const styles = theme => ({
  button: {
    boxShadow: theme.bigShadow,
    borderRadius: theme.space?.xs,
    "&:hover .arrow": {
      transform: "translateX(5px)"
    }
  },
  icon: {
    transition: "transform .35s"
  },
  noShadow: {
    boxShadow: "none"
  }
});

export default styles;
