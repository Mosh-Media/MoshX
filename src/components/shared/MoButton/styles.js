const styles = theme => ({
	button: {
		padding: theme.spacing(2),
		color: "black",
		fontSize: "20px",
		boxShadow: "0 24px 24px -18px rgba(69,104,129,.33), 0 9px 45px 0 rgba(114,119,160,.12)",
		backgroundColor: "#daff81",
		borderRadius: "3px",
		"&:hover": {
			backgroundColor: "#daff81",
		}
	}
});

export default styles;