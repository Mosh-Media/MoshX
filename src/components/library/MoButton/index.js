import React from "react";

import ArrowForwardIcon from "@material-ui/icons/ArrowForward";
import Button from "@material-ui/core/Button";
import styles from "./styles";
import withStyles from "@material-ui/core/styles/withStyles";

const MoButton = ({ classes, handleButtonClick, text }) => {
  return (
    <Button className={classes.link} onClick={() => handleButtonClick()}>
      {text}
      <ArrowForwardIcon
        className="arrow"
        style={{ fontSize: "1em", paddingLeft: "4px" }}
      />
    </Button>
  );
};

export default withStyles(styles)(MoButton);
