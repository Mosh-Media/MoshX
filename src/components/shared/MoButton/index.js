import React from "react";

import ArrowForwardIcon from "@material-ui/icons/ArrowForward";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";
import styles from "./styles";
import withStyles from "@material-ui/core/styles/withStyles";

const MoButton = ({ classes, handleClick, href, text }) => {
	return (
		<Link to={href ? href : "#"} style={{textDecoration:"none"}}>
			<Button
				color="primary"
				className={classes.button}
				onClick={() => (href ? "" : handleClick())}
				variant="contained"
			>
				{text}
				<ArrowForwardIcon />
			</Button>
		</Link>
	);
};

export default withStyles(styles)(MoButton);
