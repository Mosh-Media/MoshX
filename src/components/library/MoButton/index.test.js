import React from "react";
import renderer from "react-test-renderer";
import MoButton from "./index.js";

it("Should render MoButton", () => {
  const tree = renderer.create(<MoButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
