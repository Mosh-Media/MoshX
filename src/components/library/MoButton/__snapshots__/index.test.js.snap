// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoButton Component should match the snapshot 1`] = `
<MoButton
  classes={
    Object {
      "link": "MoButton-link-1",
    }
  }
/>
`;
